### ISSUE TITLE
errors when bundling

### ISSUE THREAD

**Issue by tobowers** [MikeMcl/bignumber.js#217]
I was running into the same errors as #199 and #166 . Also see issues at node-cbor too: https://github.com/hildjj/node-cbor/issues/88

I created a very minimal setup that has the same problems: https://github.com/quorumcontrol/minimal-fail-bignumber

Running: https://github.com/quorumcontrol/minimal-fail-bignumber/blob/master/cbor-bignumber/showerror.sh 

produces:

```
+ node dist/index.js
/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:4144
const MINUS_ONE = new bignumber(-1);
                  ^

TypeError: bignumber is not a constructor
    at Object.parcelRequire.1mQP.bignumber.js (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:4144:19)
    at newRequire (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:49:24)
    at localRequire (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:55:14)
    at Object.parcelRequire.b+YO.../vendor/binary-parse-stream (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:4268:16)
    at newRequire (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:49:24)
    at localRequire (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:55:14)
    at Object.parcelRequire.eVKc../utils (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:4921:17)
    at newRequire (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:49:24)
    at localRequire (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:55:14)
    at Object.parcelRequire.dsOm../commented (/Users/tobowers/code/minimal-fail-bignumber/cbor-bignumber/dist/index.js:6434:21)
```

**Comment by MikeMcl** [comment]
You have not stated how you are bundling or loading this library. Your "minimal setup" has an enormous list of dependencies and a *package-lock.json* file of nearly 7000 lines. Personally, I know nothing about *cbor* or about *parcel*. 

This library is essentially a single file (though for convenience there are three different versions: *bignumber.js*, *bignumber.mjs* and *bignumber.min.js*). There are no dependencies and the *package.json* is very simple. Loading the library is easy in both Node.js and the browser, whether using a script tag, CommonJS `require` or ES module `import`.

The issue you are having is unlikely to be solved by raising it here, and given the above I do not think it is the appropiate place for it.

**Comment by MikeMcl** [comment]
The error is caused by Parcel using the ES module version of this library, i.e. it is adding the *bignumber.mjs* file to the bundle instead of the *bignumber.js* file.

This can be seen by renaming the *bignumber.mjs* file before running

```bash
node node_modules/parcel/bin/cli.js build -t node --no-minify --bundle-node-modules index.js
```

in which case the *bignumber.js* file is instead bundled and everything is fine.

```bash
worked
```

For some reason, Parcel is not able to create the default export from the *bignumber.mjs* file. It instead exports `BigNumber` as a property of the exports object. This means that if all the occurences of

```js
require('bignumber.js')
```

in the bundle, *dist/index.js*, are replaced with 

```js
require('bignumber.js').BigNumber
```

the bundle will also work as expected.

**Comment by MikeMcl** [comment]
This [line](https://github.com/parcel-bundler/parcel/blob/master/packages/core/parcel-bundler/src/Resolver.js#L296) in the Parcel source code shows that when it parses the *bignumber.js/package.json* it will use the file pointed to by the *module* field in preference to that of the *main* field, and this does not seem to be configurable.

This means that the only way to solve this issue is for the *module* field to be removed from the *bignumber.js/package.json*.

Do I want to do this because of the limitations of Parcel?

**Comment by tobowers** [comment]
First, thanks so much for taking a look! Being an OSS dev is often thankless... I'm here thanking you for the project itself and for looking.

My guess is this will end up being a babel.js bug as both webpack and parcel suffer from the same problem here.

**Comment by MikeMcl** [comment]
With webpack, I think there are workarounds such as adding the following to the options object so that the *main* field takes precedence:

```
resolve: {
  mainFields: ["main", "module"]
}
```

I'm tempted to remove the *module* field but I can't remember exactly why I included it in the first place - probably because it was requested. 

[webpack #7482](https://github.com/webpack/webpack/issues/7482)

[decimal.js #59](https://github.com/MikeMcl/decimal.js/issues/59)

**Comment by MikeMcl** [comment]
I have decided to remove the non-standard `module` field from the *package.json*, which should solve this issue. I only added it for the benefit of bundlers, yet it seems to cause more problems with bundling then it solves.

I note that parcel bundling seems to work okay irrespective of whether consumers use `require` or `import`, as long as it is the commonjs *bignumber.js* that is bundled rather than the es6 module *bignumber.mjs*, i.e. as long as there is no `module` field pointing to an es6 module.

For my reference, the rationale for currently having an extensionless  `main` field in the *package.json* and including both a commonjs and es6 version of the library in the root directory, can be found at the following links:

[4.4. Shipping both ESM and CJS](https://github.com/nodejs/node-eps/blob/master/002-es-modules.md)

[Native ES Modules in NodeJS](https://medium.com/@giltayar/native-es-modules-in-nodejs-status-and-future-directions-part-i-ee5ea3001f71)
